### PR TITLE
Setting default flavour to caasp

### PR DESCRIPTION
**Pull request by robdaemon**
Since the openSUSE flavour does not currently work, it makes more sense
for the default in this repo to be the working, supported platform.

### CODE REVIEW

File: contrib/libvirt/README.md [hunk 1]
```diff
@@ -32,16 +32,16 @@ Some weird things:
   supposed to be on the same directory as your local `k8s-terraform` copy. If
   that's not the case, you can provide the `SALT_PATH` environment variable.
 - The path to the image to be used is taken from the `IMAGE_PATH` environment
-  variable. If this is not defined, an openSUSE image will be downloaded and used.
+  variable. If this is not defined, an SUSE CaaSP 1.0 image will be downloaded and used.
```
Comment (ereslibre) [inline contrib/libvirt/README.md]: I'd leave the version out...

File: contrib/libvirt/README.md [hunk 2]
```diff
@@ -32,16 +32,16 @@ Some weird things:
...
 - All nodes are given 2GiB of RAM memory by default. We consider this to be a
   good default for regular deployments. You can change that with the
   `MASTER_MEMORY` and `MINION_MEMORY` environment variables.
 - We are using 2 minion nodes. You can change this with the `MINIONS_SIZE`
   environment variable.
 - If you want to run the terraform provisioning in debug mode, you can export
   the `TF_DEBUG` environment variable set to any value.
-- By default all images used are openSUSE. This can be tweaked with the
-  `FLAVOUR` environment variable, but for now there are no other supported platforms.
+- By default all images used are SUSE CaaSP 1.0. This can be tweaked with the
```
Comment (ereslibre) [inline contrib/libvirt/README.md]: I'd leave the version out here too...

**Comment by robdaemon** [comment]
Took the 1.0 out

**Comment by robdaemon** [comment]
I thought about that, but since it doesn't work right now, I didn't want to suggest it as an option.

**Comment by ereslibre** [comment]
While it doesn't work right now I'd say we still want to document it. The idea is that it should be possible to use `opensuse`, despite we aren't centering our efforts on it right now...

**Comment by PI-Victor** [comment]
not sure when we decided this, but i disagree with making caasp the main flavor. when i read @cyphar's email he was suggesting to get into the habit of treating openSUSE as upstream to our project and not the other way around.
this defies the whole purpose of making it open source. besides https://github.com/kubic-project/terraform/issues/42 being fixed, what else is needed to make openSUSE work?

**Comment by ereslibre** [comment]
IMHO it's not about making openSUSE work, it's about having to maintain different scenarios. When the Beta 2 release was in place lots of things changed because we were developing with `opensuse` and then we changed stuff to work on `caasp`. After this `opensuse` was broken for a week (and it still is).

Don't get me wrong, I want to be able to support opensuse (of course, why not?). Given the humanpower we got we need to center our efforts on something specific, and given that it is a good idea to develop as closer as possible to production we thought we could center our development efforts on `caasp`, but keeping an eye on `opensuse`. This basically means: develop on a daily basis with `caasp` (close to production), and test that `opensuse` X times a week/whatever. I guess that once that we have been able to remove lots of divergencies between `opensuse` and `caasp` it will make sense to revisit this issue... Until then I think it's just too much to try to develop on `opensuse` without expecting surprises when everything gets baked for `caasp` on further releases.

- We have migrated [`public/private` scheme to production manifests](https://github.com/kubic-project/caasp-container-manifests/pull/34).
- We will remove manifests from velum [I am working on this ATM], and use the production ones with some substitutions.
- `OpenSUSE` will use docker hub containers (opensuse based), while `caasp` uses `sles12` docker images (delivered by `container-feeder` now, by our own registry in the future).
- https://github.com/kubic-project/terraform/issues/42 as you said
- Salt states are OS agnostic (`pkg.installed` will just do nothing on `caasp` if the package is already installed, will install it on `opensuse`)
- Probably lots of other small details

Just my 2 cents. It makes sense for now, and nothing makes me more sad that remove the open source default; but I think it makes sense until we can revisit this and we have been able to remove some divergences between `caasp` and `opensuse`.

**Comment by robdaemon** [comment]
This wasn't my favorite change to make either, but it aligns with our desire to do MicroOS first, openSUSE next for the short-term. I was attempting to lower the barrier for us to run and contribute.

I'd love to leave openSUSE the default, but we need to put the effort into making it viable.

**Comment by inercia** [comment]
We had a meeting & discussion about this and we stated that our main priority is CaaSP. It would be good to support openSUSE but we will not spend time doing that: if the community wants to do all the changes necessary for running this on openSUSE it will be fantastic and it will have our full support, but it is a task that we cannot do by ourselves.

**Pull request merged**